src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/LifeCycle/LifeCycleEffect.shared.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.Effects
+{
+	public class LifeCycleEffect : RoutingEffect
+	{
+		public event EventHandler Loaded;
+
+		public event EventHandler Unloaded;
```

Comment:
I propose adding corresponding commands as well.

Also, would be great to make this effect attachable (similar to TouchEffect)

```xaml
<StackLayout xct:LifecycleEffect.UnloadedCommand="{Binding Command}">
```

Reply:
> Also, would be great to make this effect attachable (similar to TouchEffect)

For commands yes, I'm not sure if we can do it for events

Reply:
> > Also, would be great to make this effect attachable (similar to TouchEffect)
> 
> For commands yes, I'm not sure if we can do it for events

I think we can't. But command would work

Reply:
@AndreiMisiukevich I was thinking about the syntax and since we can't do attached props. for events, I think that is a bad idea to make it for Commands, I can image a syntax similar to the `GestureRecognizers`

```xml
<TapGestureRecognizer Tapped="This is an event"  Command = ""/>
```

So that way the effect will be

```xml
<xct:LifecycleEffect Loaded="Event"  LoadedCommand="Command" />
```

Thoughts?

Reply:
That's fine

Reply:
Thinking in MVVM, I find it interesting to add commands.

Reply:
About the command, and MVVM. On `event` I provide the control as `sender`, that way the user can perform actions per control. If we add Command would be good (or necessary) to provide the `CommandParameter`, and it should be the control and that is against the MVVM because the VM will have a reference from the View (e.g. the control parents).

Thoughts?

Reply:
Could you integrate it into Shell? 
I imagine calling it this way: 
Shell.Current.CurrentPage.Loaded
Shell.Current.CurrentPage.Unloaded